identity-core/src/common/timestamp.rs

```diff
@@ -37,10 +37,24 @@ impl Timestamp {
   /// fractional seconds truncated.
   ///
   /// See the [`datetime` DID-core specification](https://www.w3.org/TR/did-core/#production).
+  #[cfg(not(all(target_arch = "wasm32", not(target_os = "wasi"), feature = "wasm")))]
   pub fn now_utc() -> Self {
     Self(truncate_fractional_seconds(OffsetDateTime::now_utc()))
   }
 
+  /// Creates a new `Timestamp` with the current date and time, normalized to UTC+00:00 with
+  /// fractional seconds truncated.
+  ///
+  /// See the [`datetime` DID-core specification](https://www.w3.org/TR/did-core/#production).
+  #[cfg(all(target_arch = "wasm32", not(target_os = "wasi"), feature = "wasm"))]
```

Comment:
I think I would prefer having that fixed in the wasm bindings rather than in core. Any advantage of having it here? I guess there is this 9999AD invariant that you want to uphold, which is easier to state here because of the "proximity" to the other code that upholds that invariant. But aside from that, I would prefer having binding-specific code in the bindings directory, otherwise this sets a sort of precedent on where binding-specific issues are fixed.

Reply:
It needs to be in `identity-core` because `IotaDocument::new()` calls `Timestamp::now_utc()` internally (and probably in other places in the code too). Changing only external `WasmTimestamp::now_utc()` calls in the Wasm bindings would not fix the current problem as far as I know.

Reply:
Ah thanks, that makes sense.

Reply:
@PhilippGackstatter when I find a bit more time I will see if I can make the `time` crate compatible with `wasm-bindgen` itself and then we can remove this code. For now I think it is OK to have it here for the reasons @cycraig suggested.